TAScheduler/Management/CourseManagement.py

```diff
@@ -58,6 +58,29 @@ def editCourse(Id, courseName, courseLocation, courseDays, courseHours, courseIn
 
         return "The course was successfully edited"
 
+#Find course method for editing courses
+    @staticmethod
+    def findCourse(courseID):
+        if not (courseID == 0):
+            CourseManagement.__inputErrorCheck(courseID=courseID)
+            try:
+                course = Course.objects.get(courseID=courseID)
+            except Course.DoesNotExist:
+                course = None
+
+            if course is None:
+                raise TypeError("This ID does not exist")
+        elif not (courseID== ""):
+            try:
+                course = Course.objects.get(courseID=courseID)
+            except Course.DoesNotExist:
+                course = None
+
+            if course is None:
+                raise TypeError("This username does not exist")
```

Comment:
```suggestion
                raise TypeError("This ID does not exist")
```
courseID should never be a string. I believe you left in some vestigial code from findUser here where it looks for a username. Also, there is no reason for course to be set to None and then later check if it is none. You can simply raise the error in the except block.